### ISSUE TITLE
Feature: Scoring for individual releases

### ISSUE THREAD

**Issue by nadgowdas** [ossf/scorecard#1985]
**Is your feature request related to a problem? Please describe.**

Currently, scorecard runs against the `repository` settings (typically `main` branch or commit-sha). Malicious developer can circumvent the scoring by implementing bad practice between two runs as shown below:

<img width="1214" alt="Screen Shot 2022-06-06 at 9 17 16 AM" src="https://user-images.githubusercontent.com/8685412/172168136-4dc5e123-26cd-4c3d-9f11-bfa5a16b7038.png">

This is possible, because SCM/github do not maintain immutable ledger (like commit hash) of repository setting changes. 
Therefore, the scoring at the repository *may* not be applicable to the individual releases from that repository.

Can we add support for scoring individual releases ?

**Describe the solution you'd like**
For a given release of a given repo, we discover all the PRs/commits that went into that release (since the baseline of previous release or any given release). Then collect the aggregated metrics for a given release that indicates whether all PRs/commits were peer reviewed, how many unique developers contributed in this release, whether there were any commits directly to the `main` branch, what "kinds" of changes went into the release (based on labels of PRs and linked issues), etc.

<img width="257" alt="Screen Shot 2022-06-06 at 9 25 09 AM" src="https://user-images.githubusercontent.com/8685412/172169348-99ec96e0-3f7f-4cfb-9200-130478b6f836.png">

**Additional context**
There is an open-source tool I have built to illustrate such use-cases. The tool is called [gauge](https://github.com/IBM/gauge).

E.g. when I ran it against open-source python package `flask` it reports as follows. 

```
./gauge package -p flask -e python -t 2.1.1 -r https://github.com/pallets/flask
------------------------------------------------------------------------------
Gauge Report for package `flask`
------------------------------------------------------------------------------
Release Measures:
	Current version: 2.1.1
	Latest version: 2.1.2
	Release lag (versions): 1
	Release lag (days): 28 days
--------------------------------------------------------------------------------
		Recommended update
		 Version - 2.1.2
		 Release Time - 2022-04-28 17:48:24 +0000 UTC
		 Num of unique contributors - 7
		 Num of unique reviewers - 0
		 Non peer reviewed changes - 0
		 Num of zombie commits - 15
		 Change annotations - ['docs','typing','testing']
--------------------------------------------------------------------------------
```

The core motivation behind such feature would be to allow users to evaluate respective release versions for OSS artifacts before consuming and provide them some insights to make an informed decision to upgrade.

**Comment by azeemshaikh38** [comment]
A wild idea - what if we collaborate with Dependabot/Renovatebot folks here so that every version bump PR by these tools provides a summarized view of diff between new version and existing one? It'll be super impactful and will raise OSS-wide awareness if all OSS developers who use these tools today start thinking about security of their dependencies. 

@rarkins @josepalafox what do you folks think?

**Comment by rarkins** [comment]
For npm releases, Renovate currently provides a link in PRs to a diff of the two release artifacts (ie comparing npm tarballs). You can't trust that the source on GitHub is what was used to create the tarballs on npm

**Comment by nadgowdas** [comment]
@azeemshaikh38 I liked the idea, would be super useful.

I think in addition to providing link to PR diffs, if we can curate some of the information and provide developers with actionable insights, they can easily put it in their automated workflows.

**Comment by azeemshaikh38** [comment]
@rarkins makes sense for npm but this could be useful for non-npm ecosystems like Golang? What do you think about adding such diffs in Renovatebot?

**Comment by rarkins** [comment]
We plan to add more over time. Golang is not the best example because it's one of the few where "what you see is what you get" (on github). Registries where the artifact may differ from the source - for both legitimate and illegitimate reasons - are the most challenging for the user and benefit from this type of diff.
